hspec-discover/test/Test/Hspec/Discover/SortSpec.hs

```diff
@@ -0,0 +1,49 @@
+module Test.Hspec.Discover.SortSpec (main, spec) where
+
+import           Helper
+
+import           Data.List (sortBy)
+import           Data.Ord (comparing)
+
+import           Test.Hspec.Discover.Sort
+
+main :: IO ()
+main = hspec spec
+
+
+spec :: Spec
+spec = do
+  describe "naturalSortKey" $ do
+    context "compares" $ do
+      it "reflexively" $ do
+        "" <=> "" `shouldBe` EQ
+        "Hello2World" <=> "Hello2World" `shouldBe` EQ
+        "123Hello456" <=> "123Hello456" `shouldBe` EQ
+        "" <=> "" `shouldBe` EQ
+      it "empty first" $ do
+        "" <=> "Hello2World" `shouldBe` LT
+        "Hello2World" <=> "" `shouldBe` GT
+      it "numbers first" $ do
+        "Hello2World" <=> "Hello World" `shouldBe` LT
+        "Hello World" <=> "Hello2World" `shouldBe` GT
+      it "string segments" $ do
+        "Hello2World9" <=> "Hello2World!0" `shouldBe` LT
+        "Hello2World!0" <=> "Hello2World9" `shouldBe` GT
+      it "numeric segments" $ do
+        "3.1.415" <=> "3.14.15" `shouldBe` LT
+        "3.14.15" <=> "3.1.415" `shouldBe` GT
+      it "numeric ties by string" $ do
```

Comment:
I think this is the right thing to do :clap:

Many other implementations for natural sorting miss this case and as a result assume that e.g. `00` and `0` are equal.

Reply:
It was only after writing this test that I realized I wanted to add it to the comparator :)